cumulusci/tasks/github/access_report.py

```diff
@@ -0,0 +1,264 @@
+from typing import Set
+import datetime
+import os
+import csv
+from itertools import chain
+
+
+from jinja2 import Environment, FileSystemLoader
+
+from cumulusci.tasks.github.base import BaseGithubTask
+from cumulusci.core.utils import process_list_arg
+
+
+class OrganizationReport(BaseGithubTask):
+    task_options = {
+        "organization": {
+            "description": "The name of the Github organization",
+            "required": True,
+        },
+        "output": {
+            "description": "The file where the report should be generated",
+            "required": True,
+        },
+        "template": {
+            "description": "The path to a custom jinja2 template file for rendering the report",
+        },
+        "repos": {
+            "description": "A comma-separated list of Repo names to report for (not including org-name)",
+        },
+        "additional_info_csv": {
+            "description": "The path to a CSV file. "
+            "The CSV must have column headings in the first row. "
+            "One of the column headings must be 'Github Username'. "
+            "The other columns will be added to the Member report."
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+        self.template = self.options.get("template") or os.path.join(
+            "cumulusci",
+            "tasks",
+            "github",
+            "report_templates",
+            "github_access_report.html",
+        )
+        repos = self.options.get("repos")
+        self.repos = set(process_list_arg(repos)) if repos else None
+        self.additional_info_csv = self.options.get("additional_info_csv")
+
+    def _run_task(self):
+        org = self.github.organization(self.options["organization"])
+        if self.additional_info_csv:
+            member_extra_info = self._parse_member_extra_info(self.additional_info_csv)
+            extra_fields = list(next(iter(member_extra_info.values())).keys())[1:]
```

Comment:
I spent about a minute figuring out what this does. I think it'd be a lot clearer if _parse_member_extra_info grabbed the fieldnames while iterating through the data and returned a 2-tuple of (extra_fields, member_extra_info)